src/integrations/strapi/getAllSlugsFromStrapi.ts

```diff
@@ -18,33 +19,52 @@ const getAllSlugsFromStrapi = async <T extends LocalizedEntity<'slug'>>(
   locales: string[],
   { filters = {} }: Options = { filters: {} }
 ): Promise<Slug[]> => {
-  const params: Record<string, any> = {
-    locale: 'all',
-    'pagination[pageSize]': STRAPI_DEFAULT_PAGE_SIZE,
-    filters,
-  };
-
-  const { data } = await strapiClient.get<IStrapiResponse<IStrapiData<T>[]>>(
-    path,
-    { params }
-  );
+  let allSlugs: Slug[] = [];
+
+  const allLocales = await getAvailableLocalesFromStrapi();
+
+  const slugPromises = allLocales.map(async (locale) => {
+    const params: Record<string, any> = {
+      locale,
+      'pagination[pageSize]': STRAPI_DEFAULT_PAGE_SIZE,
+      filters,
+    };
+
+    const { data } = await strapiClient.get<IStrapiResponse<IStrapiData<T>[]>>(
+      path,
+      { params }
+    );
```

Comment:
```suggestion
  const slugPromises = allLocales.map((locale) => {
    const params: Record<string, any> = {
      locale,
      'pagination[pageSize]': STRAPI_DEFAULT_PAGE_SIZE,
      filters,
    };

    const { data } = strapiClient.get<IStrapiResponse<IStrapiData<T>[]>>(
      path,
      { params }
    );
```

Do you need those async/await if you only return a promise from the .map?

Reply:
Yes because for the `return data.data.map((page)=>...)` I need the data from the strapiClient to continue.